src/components/features/SignUpForm/SignUpForm.tsx

```diff
@@ -0,0 +1,31 @@
+"use client";
+import { useCallback, useState } from "react";
+import { Button } from "../Button/Button";
+import { InputField } from "../InputField/InputField";
+import Style from "./SignUpForm.module.scss";
+import { formSchema } from "./form.schema";
+import { FormState } from "./form.types";
+import { getDefaultFormValues } from "./form.utils";
+
+export function SignUpForm() {
+  const [formState, setFormState] = useState<FormState>(getDefaultFormValues);
+  const updateFormValues = useCallback((newFormValue: Partial<FormState>) => {
+    setFormState((previousFields) => ({ ...previousFields, ...newFormValue }));
+  }, []);
+
+  return (
+    <form className={Style.container}>
+      {formSchema.map((field) => (
```

Comment:
Please add validations for password, username and email. If there are errors change the status of the input to show which input is incorrect and show messages that describe the reason of the error. Also if there is an error please disabled the submit button